File: scripts/reporting/templates.py
import os

# Base directory for the assets
ASSETS_DIR = os.path.join(os.path.dirname(__file__), 'assets')

def read_asset(filename):
    """Utility function to read a static asset file."""
    with open(os.path.join(ASSETS_DIR, filename), 'r', encoding='utf-8') as f:
        return f.read()

def get_html_template():
    return read_asset('template.html')

def get_css_asset():
    return read_asset('style.css')

def get_js_asset():
    return read_asset('app.js')

def get_rca_section():
    return read_asset('rca_section.html')

File: scripts/reporting/templates.js
import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Base directory for the assets
const ASSETS_DIR = path.join(__dirname, "assets");

/**
 * Utility function to read a static asset file.
 * @param {string} filename
 * @returns {string}
 */
function readAsset(filename) {
  const filePath = path.join(ASSETS_DIR, filename);
  return fs.readFileSync(filePath, "utf-8");
}

export function getHtmlTemplate() {
  return readAsset("template.html");
}

export function getCssAsset() {
  return readAsset("style.css");
}

export function getJsAsset() {
  return readAsset("app.js");
}

export function getRcaSection() {
  return readAsset("rca_section.html");
}
